manager/controllers/app/select_modules.go

```diff
@@ -104,6 +116,16 @@ func (p *PlotterGenerator) validSolutions(item *DataInfo, solutions []Solution,
 	return validPaths
 }
 
+// Return true if bucket allocation is needed. Otherwise return false.
+func (p *PlotterGenerator) bucketAllocationIsNeeded(item *DataInfo) bool {
```

Comment:
The name of the function is misleading, it is not as if bucket allocation is always required for a specific item.
Since you are using this function only once, I suggest you replace the function call with the actual statement:
```
item.Context.Flow != taxonomy.WriteFlow ||
		item.Context.Requirements.FlowParams.IsNewDataSet
```

Reply:
Fixed. Thanks